specs/v0.1.0.md

```diff
@@ -0,0 +1,102 @@
+
+This project will provide the following features:
+
+- Generate a ‘recipes’ database and feed initial recipes to it.
+- Create/Update/Delete/Read recipes and their ingredients.
+- Rate recipes.
+- Randomly suggest a list of recipes for the week.
+
+What this project will provide in a future version:
+
+- Adding new recipes to the database  based on preferences.
+- Once the recipes are approved by the user, generate a shopping list of all the ingredients needed to make them.
+
+We will to create a SQLite database storing our favourite recipes and issue a selection of 10 recipes for the week via one REST API endpoint. The server will be implemented in Python 3.6.5.
+
+## Project organization
+
+### Poetry
+
+We will use [Poetry](https://poetry.eustace.io/) to structure our source code, manage dependencies, control the Python environment. `poetry` also offers a centralized way of executing several scripts like so: `poetry run <script>`.
+
+We will provide the following script:
+
+ - `api`: defines the functions that will be queries to the database using the sqlite3 module.
+ - `run`: runs the previously defined functions to obtain the recipe list and the final ingredient lists.
+ - `test`: will be testing our script
+
+
+### Packages
+
+The project will be composed of the following packages:
+ - `shopping_list`: Contains the code for the API that will generate the suggested recipes.
+ - `tests`: Contain all the unit tests.
+
+## Database
+
+### Schema
+
+#### Recipes
+| Column        | Type   | Attributes    | Description |
+| ------------- | ------ | ------------- | ----------- |
+| id            | `INT`  | `PRIMARY KEY` | recipe id |
+| name          | `TEXT` | `NOT NULL`    | name of the recipe|
+| rating        | `INT`  | `NOT NULL`    | grade of the recipe (max value is 5)|
+| total_time    | `INT`  | `NOT NULL`    | preparation time + cooking time|
+| ingredient_id | `INT`  | `FOREIGN KEY` | foreign key to the ingredients table|
+
+
+
+#### Ingredients
+
+| Column        | Type   | Attributes    | Description |
+| ------------- | ------ | ------------- | ----------- |
+| id            | `INT`  | `PRIMARY KEY` | ingredient id |
+| name          | `TEXT` | `NOT NULL`    | name of the ingredient|
+| amount_id     | `INT`  | `FOREIGN KEY` | foreign key to the amounts table|
+
+
+
+#### Amounts
+
+| Column        | Type   | Attributes    | Description |
+| ------------- | ------ | ------------- | ----------- |
+| id            | `INT`  | `PRIMARY KEY` | amount id |
+| name          | `INT`  | `NOT NULL`    | quantity  |
+
+
+### Seeding
+
+This will be done using SQLite DB Browser because we can perform tasks such as creating databases, searching them and editing the data in a more visual way.
+
+In a future version we will provide a script to do that automatically.
+For a start, we will add about 13 recipes to the database, the number of recipes will grow over time.
```

Comment:
13 is a pretty exact number 😆 Minor, but if you intend to add "exactly" 13 recipes you might want to remove "about" If you are not sure it will be 13, then something like "about a dozen recipes" will make it!.